lib/enum/value.rb

```diff
@@ -0,0 +1,144 @@
+module Enum
+  class Value
+    
+    # TODO: Perhaps stored_value should be renamed unsafe_value?
+    
+    include Comparable
+
+    attr_reader :stored_value, :error
+    
+    class << self
+      attr_accessor :default_value, :suppress_read_errors, :klass
+    end
+    
+    def self.inherited(subclass)
+      # Value subclass settings and options
+      #   @default_value => <:ERROR|:ANY|:something|nil>
+      #   @suppress_read_errors => <true|false>
+      subclass.suppress_read_errors = false
+      subclass.default_value = :ERROR
+      super
+    end
+    
+    # Combined getter/setter for 'default_value'.
+    def self.default_value(*args)
+      # Can't use .any? because [nil].any? is false (ruby 2.4),
+      # and nil is a valid argument here.
+      # [nil].empty? is also false, which is what we want.
+      if args.empty?
+        @default_value
+      else
+        @default_value = args[0]
+      end
+    end
+    
+    # Combined getter/setter for 'suppress_read_errors'.
+    def self.suppress_read_errors(*args)
+      if !args.empty?
+        @suppress_read_errors = args[0]
+      else
+        @suppress_read_errors
+      end
+    end
+    
+    # Load a primitive (symbol, string, integer) into new enum Value instance,
+    # taking into consideration enum constraints, default_value setting.
+    # Returns frozen Value instance.
+    def initialize(raw_val)
+      begin
+        @stored_value = klass[raw_val].to_sym.freeze
+      rescue Enum::TokenNotFoundError => _error
+        @error = _error.freeze
+        case
+        when self.class.default_value == :ERROR
+          raise _error
+        when self.class.default_value == :ANY
+          @stored_value = raw_val.freeze
+        else
+          @stored_value = self.class.default_value.freeze
+        end
+      end
+      self.freeze
+      self
```

Comment:
no need to return `self` from an initializer.

Reply:
Oh right... I probably thought I was working on `.new` 😅. Will fix.